packages/sign-client/src/controllers/engine.ts

```diff
@@ -1462,8 +1467,11 @@ export class Engine extends IEngine {
         params.expiryTimestamp || calcExpiry(ENGINE_RPC_OPTS.wc_sessionPropose.req.ttl);
       const proposal = { id, pairingTopic: topic, expiryTimestamp, ...params };
       await this.setProposal(id, proposal);
-      const hash = hashMessage(JSON.stringify(payload));
-      const verifyContext = await this.getVerifyContext(hash, proposal.proposer.metadata);
+      const verifyContext = await this.getVerifyContext({
+        attestationId: attestation,
+        hash: hashMessage(JSON.stringify(payload)),
```

Comment:
Nit I'd consider running some validation against the payload.

Reply:
the payload is already validated at the top of the method